### DISCUSSION TITLE
I can't add additional parameters when adding or updating an item with the Ecommerce Framework's Cart.

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#16249] [Q&A / Support] [answered]
![image](https://github.com/pimcore/pimcore/assets/70108013/13fe2efc-32fd-4185-af90-7b04c18053fc)

I need to store additional information about my products when adding them to the cart, and while reading the accepted arguments for adding an item or updating it, I found that I can pass additional parameters. This would be very helpful for me because when generating an order, I need to retrieve the extra parameters that I added to store them in the order's items. However, I don't see any method to retrieve these parameters; I can only add a comment and retrieve it using the getComment() method when getting the item. Could you please tell me how I can retrieve the params? I am adding them like this:

`$cart = $this->cartService->getCart();
        $product = Product::getById(63);
        $cart->addItem( $product, 8,$product->getId(),true,['storeId' => 2] ,[],'test-comment');
        $cart->save();`

**Comment by AlephLau** [comment]
@mattamon When you have the chance, do you think you could clarify for me, please?

**Comment by mattamon** [comment, marked as answer]
@AlephCode 
In this case I am sorry to inform you that it seems that what you are trying to do is not possible.
If you checkout the update method https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/CartManager/AbstractCart.php#L117 the params are never used or saved anywhere which is unfortunate.
I have no idea what the original intention here was on how to save those params but maybe @dvesh3 can help me out here.

My best guess would be to use the CheckoutData since you can store data there easily and you have them available while creating the order.

**Reply by AlephLau** [reply]
Don't worry about it. I want to thank you for always answering my questions, you are very kind. In the end, what I did was leverage the 'comment' parameter to store the information there as JSON, but I'll take a look at 'checkout Data' when I have time. Thanks again, @mattamon
